### DISCUSSION TITLE
Resampling algorithm and Padding

### DISCUSSION THREAD

**Discussion by vincentsarago** [cogeotiff/rio-tiler#650] [General]
Following discussion from #647 here is a small analysis of the impact of the resampling and padding 

https://gist.github.com/vincentsarago/ff73ea5dbf128949d951851cc8454b4f

Tile within the dataset zoom range: 

<img width="1008" alt="Screenshot 2023-10-23 at 11 46 13 PM" src="https://github.com/cogeotiff/rio-tiler/assets/10407788/4ea56973-cbf1-4e04-b42f-9019d86e6fa7">

☝️ There isn't any highly visible edge artefacts but we know the data is somehow slightly different (but nothing visually noticeable with or without padding)

Overzoom (tile at zoom > than dataset max resolution)

<img width="1006" alt="Screenshot 2023-10-23 at 11 46 49 PM" src="https://github.com/cogeotiff/rio-tiler/assets/10407788/118e6b8a-e97d-4b24-92b0-710254e281f0">

☝️ We can clearly see edges artefact and disappear when using `padding=2`. Also noticing that padding has no influence when using `nearest` resampling.

the TLDR, padding is useful mostly when overzooming. When in the dataset's zoom range there is not highly visible edges so padding do not have huge impact.
